Lib/shutil.py

```diff
@@ -529,9 +529,11 @@ def _copytree(entries, src, dst, symlinks, ignore, copy_function,
             else:
                 # Will raise a SpecialFileError for unsupported file types
                 copy_function(srcobj, dstname)
-        # catch the Error from the recursive copytree so that we can
-        # continue with other files
+        except SameFileError as err:
+            errors.append((srcname, dstname, str(err)))
         except Error as err:
+            # catch the Error from the recursive copytree so that we can
+            # continue with other files
```

Comment:
This comment should go before the `except Error` like before, shouldn't it?

Reply:
I moved it further up.